owncloudApp/src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```diff
@@ -274,24 +274,6 @@ private Bitmap doOCFileInBackground() {
 
                 int px = getThumbnailDimension();
 
-                if (file.isDown()) {
```

Comment:
What happens when there's no connection? Are the already downloaded thumbnails not going to be shown in that case?

Reply:
Steps to show a thumbnail:
1 Check if thumbnail is available in cache
2 a) If thumbnail is not available -> Ask preview API
2 b) If needsUpdateThumbnail flag is active* -> Ask preview API
2 c) thumbnail available and flag not active -> Show cached thumbnail
3 a) If preview API request is not successful, default image
3 b) If preview API request is successful, save thumbnail and show it

*needsUpdateThumbnail is active if modification time differs between remote and local one. And you need connection to set it up

CC/ @jesmrec

Reply:
That's it. If there's no connection, thumbnail will not be fetched/updated (expected behaviour). It can cause a side-effect when image is locally edited and no connection is available. In that case, thumbnail and image will not be the same. As idea, it should be removed (not sure if feasible) waiting for connection to upload the new image version and fetch the new thumbnail. WDYT?

Reply:
Ok, i see, at that point, i think we could add a new step when preview API request fails. At that point, instead of showing a default image, we could try to generate a thumbnail as we we doing before

Reply:
Would it be replaced again in the next refresh?